yarn-project/prover-client/src/orchestrator/orchestrator.ts

```diff
@@ -722,3 +742,11 @@ export class ProvingOrchestrator {
     );
   }
 }
+
+function extractAggregationObject(proof: Proof, numPublicInputs: number): Fr[] {
+  const buffer = proof.buffer.subarray(
+    Fr.SIZE_IN_BYTES * (numPublicInputs - AGGREGATION_OBJECT_LEN),
+    Fr.SIZE_IN_BYTES * numPublicInputs,
+  );
+  return BufferReader.asReader(buffer).readArray(AGGREGATION_OBJECT_LEN, Fr);
+}
```

Comment:
This is based on this snippet from barretenberg

https://github.com/AztecProtocol/aztec-packages/blob/432cec020ed1bb73adcc1d6be8159609c22d328e/barretenberg/acir_tests/sol-test/src/index.js#L175-L183

The public inputs are extracted from the proof_as_fields object but then the rest of the proof is extracted by removing the equivalent bytes from the binary proof. Looks to me like the public inputs are already encoded as field elements in the binary proof so I can just read what I need from them.